lib/Parse/ParseIfConfig.cpp

```diff
@@ -47,6 +47,24 @@ Optional<PlatformConditionKind> getPlatformConditionKind(StringRef Name) {
     .Default(None);
 }
 
+/// Get platform condition name from PlatformConditionKind.
+static StringRef getPlatformConditionName(PlatformConditionKind Kind) {
```

Comment:
If we're going this far, we should probably create a `PlatformConditionKind.def` file and include it here and in the other places where we list out all the platform conditions. [Here's a commit from a few months ago where I added a similar .def file.](https://github.com/apple/swift/commit/31aeb64efa23cedb7ea66b45f7ff7bcfa5a1a4c1) In this case, I think you'll want the entries to look something like:

```
#ifndef PLATFORM_CONDITION
#define PLATFORM_CONDITION(LABEL, IDENTIFIER)
#endif
#ifndef PLATFORM_CONDITION_
#define PLATFORM_CONDITION_(LABEL, IDENTIFIER) PLATFORM_CONDITION(LABEL, "_" ## IDENTIFIER)
#endif

PLATFORM_CONDITION(OS, "os")
PLATFORM_CONDITION(Arch, "arch")
PLATFORM_CONDITION_(Endianness, "endian")
PLATFORM_CONDITION_(Runtime, "runtime")
PLATFORM_CONDITION(CanImport, "canImport")
PLATFORM_CONDITION(TargetEnvironment, "targetEnvironment")

#undef PLATFORM_CONDITION
#undef PLATFORM_CONDITION_
```

(The separate `PLATFORM_CONDITION_` definition lets you treat the internal and public conditions differently.)

Reply:
Ok, got it.